src/commands/generate/typeGenerator.ts

```diff
@@ -66,32 +66,57 @@ export function generateType(
 export async function generateEnum(definition: IEnumDefinition, simpleAst: SimpleAst): Promise<void> {
     const sourceFile = simpleAst.createSourceFile(definition.typeName);
 
-    const namespaceDefinition = sourceFile.addNamespace({
-        isExported: true,
-        name: definition.typeName.name,
-        typeAliases: definition.values.map(enumValue => ({
+    if (definition.values.length > 0) {
```

Comment:
can we dedupe the code? seems like a lot of the code in the branches is very similar

Reply:
Most of the code is actually pretty different. These are pretty much the only things that are duplicated:
```
        if (definition.docs != null) {
            variableStatement.addJsDoc(definition.docs);
        }
```
```
        sourceFile.addTypeAlias({
            isExported: true,
            name: definition.typeName.name,
```
I'm not sure it's worth deduping these things, but I can introduce functions that abstract some of this away.

I can also try unifying these code paths a bit more but I'm pretty sure it will make it less readable due to increased complexity.

What do you think?